code/modules/guardian/guardianbuilder.dm

```diff
@@ -105,10 +105,10 @@
 		return
 	switch(action)
 		if("name")
-			guardian_name = params["name"]
+			guardian_name = sanitize(params["name"])
 			. = TRUE
 		if("set")
-			switch(params["name"])
+			switch(sanitize(params["name"]))
```

Comment:
This doesn't need sanitisation since its used in a switch statement. The user input is never stored or used anywhere else except for this check.